structured/structured.proto

```diff
@@ -79,9 +79,11 @@ message TableDescriptor {
   // next_column_id is used to ensure that deleted column ids are not reused.
   optional uint32 next_column_id = 4 [(gogoproto.nullable) = false,
       (gogoproto.customname) = "NextColumnID"];
-  repeated IndexDescriptor indexes = 5 [(gogoproto.nullable) = false];
+  optional IndexDescriptor primary_index = 5 [(gogoproto.nullable) = false];
```

Comment:
I've named it index to be consistent with indexes below.
Hmmm, yeah I suppose we could make it nullable or perhaps we should just add a method IsValid() which hides the name comparison.